awx/api/views/__init__.py

```diff
@@ -1445,6 +1445,37 @@ class CredentialInputSourceSubList(SubListCreateAttachDetachAPIView):
     parent_key = 'target_credential'
 
 
+class CredentialTypeExternalTest(SubDetailAPIView):
+    view_name = _('External Credential Type Test')
+    model = models.CredentialType
+    serializer_class = serializers.EmptySerializer
+
+    def post(self, request, *args, **kwargs):
+        obj = self.get_object()
+        credential_id = request.data.get('credential', None)
+
+        test_inputs = {}
+        if credential_id is not None:
+            cred = models.Credential.accessible_objects(request.user, 'use_role').filter(id=credential_id).first()
+            if cred is None or cred.credential_type.kind != 'external':
+                return Response({}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
We wouldn't here through UI usage, but this check is for direct API usage.

Reply:
Yep, it's smart to check/limit this access.